doc/design/components.md

```diff
@@ -0,0 +1,252 @@
+
+This leads to another question regarding why we don't simply invest our project resources in helping Docker decompose the daemon. Again the answer comes back to risk management - Docker's priorities are not VMware's and, without a formal partnership, assuming the decompostion stays on a schedule that's viable for VIC deadlines is, well, risky. This is a fine line to walk as we _want_ to be good community members and, from a more selfish perspective, don't want to invest in quickly deprecated code. Our intent is to, as with the engine-api project, move to the docker and OCI components as they become semi-stable and invest resource in them at that point - how early in the life of a given component we can do that is a function of deadlines, resources and politics.
+
+
+### imagec
+
+The component follows the naming pattern introduced by OCI with 'runc' and is a docker registry client library, purely concerned with the pull/push aspects of the registry with login becoming a transitive dependency.
+
+[Current open issues relating to imagec](https://github.com/vmware/vic/labels/component%2Fimagec)
+
+
+### Port Layer
+
+The port layer presents container primitive operations, with APIs that express intent rather than the intricies of how.
+
+#### Port Layer - Execution
+
+This component deals handles management of containers such as create, start, stop, kill, etc, and is broken distinct from interaction primarily because the uptime requirements may be different, [see Interaction](#portlayer---interaction).
+
+[Issues relating to portlayer execution](https://github.com/vmware/vic/labels/component%2Fportlayer%2Fexecution)
+
+
+#### Port Layer - Interaction
+
+This component deals handles interaction with a running container and is broken distinct from execution primarily because the uptime requirements may be different.
+
+If the execution portion of the port layer is unavailable then only container management operations around creating, starting, stopping, et al are impacted, and then only for as long as the VCH is unavailable.
+
+If the interaction portions are unavailable it impacts ongoing use of interactive sessions and potentially loses chunks of the container output (unless serialized to vSphere infrastructure as an intermediate step - [container logging](#container-logging), [container base](#container-base), and [tether](#tether)) are the other components that factor into the log persistence discussion).
+
+[Issues relating to portlayer interaction](https://github.com/vmware/vic/labels/component%2Fportlayer%2Finteraction)
+
+
+#### Port Layer - Networking
+
+A Virtual Container Host presents a cluster of resource as a single endpoint, so while we must also supply transparent cross host networking, it must be there without requiring any user configuration.
```

Comment:
plural for resource "A Virtual Container Host presents a cluster of resource**s** as a single endpoint..."